src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorAdornmentManager.cs

```diff
@@ -0,0 +1,216 @@
+                    }
+                }
+            }
+
+            var map = GetSpansOnEachLine(changedSpanCollection);
+            foreach (var (lineNum, tagMappingSpanList) in map)
+            {
+                var tagMappingSpan = GetHighestOrderTag(tagMappingSpanList);
+                if (tagMappingSpan != null)
+                {
+                    TryMapToSingleSnapshotSpan(tagMappingSpan.Span, TextView.TextSnapshot, out var span);
+                    var geometry = viewLines.GetMarkerGeometry(span);
+                    if (geometry != null)
+                    {
+                        var tag = tagMappingSpan.Tag;
+                        var classificationType = _classificationRegistryService.GetClassificationType(InlineErrorTag.TagID + tag.ErrorType);
+                        var graphicsResult = tag.GetGraphics(TextView, geometry, GetFormat(classificationType));
+                        if (!_tagSpanToPointMap.TryGetValue(tagMappingSpan, out var point))
+                        {
+                            continue;
+                        }
+
+                        var lineView = TextView.GetTextViewLineContainingBufferPosition(point);
+
+                        var visualElement = graphicsResult.VisualElement;
+                        if (tag.Location is InlineErrorsLocations.HookedToCode)
+                        {
+                            Canvas.SetLeft(visualElement, lineView.Right);
+                        }
+                        else if (tag.Location is InlineErrorsLocations.HookedToWindow)
+                        {
+                            Canvas.SetLeft(visualElement, TextView.ViewportWidth - visualElement.DesiredSize.Width);
+                        }
+
+                        Canvas.SetTop(visualElement, geometry.Bounds.Bottom - visualElement.DesiredSize.Height);
+
+                        if (lineView.Right < TextView.ViewportWidth - visualElement.DesiredSize.Width)
+                        {
+                            AdornmentLayer.AddAdornment(
+                                behavior: AdornmentPositioningBehavior.TextRelative,
+                                visualSpan: span,
+                                tag: tag,
+                                adornment: visualElement,
+                                removedCallback: delegate { graphicsResult.Dispose(); });
+                        }
+                    }
+                }
+            }
+        }
+
+        private static IMappingTagSpan<InlineErrorTag>? GetHighestOrderTag(List<IMappingTagSpan<InlineErrorTag>> list)
+        {
+            foreach (var span in list)
+            {
+                if (span.Tag.ErrorType is PredefinedErrorTypeNames.SyntaxError)
+                {
+                    return span;
+                }
+            }
+
+            if (list.Count > 0)
+            {
+                return list[0];
+            }
+
+            return null;
```

Comment:
list.Where(s => s.Tag.ErrorType is PredefinedErrorTypeNames.SyntaxError).FirstOrDefault() ?? list.FirstOrDefault();

Reply:
that is much more concise, fixed.